### DISCUSSION TITLE
Meaning of placed on folder

### DISCUSSION THREAD

**Discussion by khuongviettai** [vercel/next.js#64316] [Help] [answered]
### Summary

I still don't understand what it means to name a folder like (folder), folder, [folder]. Even though I've read it on the website, I still don't understand

**Comment by SouthLink** [comment, marked as answer]
next.js uses convention-based routing, which I'll briefly describe:

1. folder: represents the normal route, which will map to the real route.
    folder path: app/blog/page.js
    Real page route: /blog

2. [folder]: represents dynamic routing, if your blog has many posts, each with an id, then you route might be:
    Folder route: app/blog/[id]/page.js
    Real page route: /blog/1

3. (folder): stands for router group, which won't map to the real route, because you might have a lot of folders with a confusing code structure and want to group those folders, you might do this.
    Folder path: app/(shop)/blog/page.js
    Real page route: /blog
